### PR TITLE
Added Keyboard Navigation

### PR DESCRIPTION
**Pull request by s-ddavydenko**
**What did you do?**
Improved accessibility by adding keyboard navigation. Features added:

-     keyboard navigation (feature)
-     focus outline (styles)
-     aria roles (aria-selected, aria-label, tabindex, listbox, option) on elements

**Why did you do it this way?**
  The appropriate aria roles and labels make the visual more understandable to screen readers which makes the sankey chart viewable to blind users and more. Keyboard navigation allows these users to actually use the sankey chart.

**Anything else I should know?**
  Updated dependencies and added unit tests

**Screenshots**
Before:
image

After:
SankeyKeyboardNavigation

### CODE REVIEW

**Comment by s-ddavydenko** [comment]
@microsoft-github-policy-service agree

**Review by zBritva — APPROVED** [review]

**Comment by zBritva** [comment]
Could you add unit tests for the new feature?

**Comment by zBritva** [comment]
image

The screen reader output in not descriptive.
And outline is not prefect for this links example. Did you consider to use custom outline? (border or custom path element to highlight the current link)

File: src/sankeyDiagram.ts [hunk 1]
```diff
@@ -1757,6 +1762,10 @@ export class SankeyDiagram implements IVisual {
                     return SankeyDiagram.createLink(link);
                 }
             )
+            .attr("role", "option")
+            .attr("tabindex", 0)
+            .attr("aria-selected", sankeyDiagramUtils.isDataPointSelected)
+
```
Comment (zBritva) [inline src/sankeyDiagram.ts:1768]: Need to add `aria-label` for desccriptive output

**Review by zBritva — CHANGES_REQUESTED** [review]

File: src/sankeyDiagram.ts (hunk 1 above)
Comment (zBritva) [reply src/sankeyDiagram.ts:1769]: Link ID is not for users, don't expose it to UI:

image

**Review by zBritva — CHANGES_REQUESTED** [review]

File: capabilities.json [hunk 2]
```diff
@@ -30,6 +30,7 @@
             "displayNameKey": "Visual_SourceLabels"
         }
     ],
+
```
Comment (MulyukovAidar) [inline capabilities.json:33]: weird

File: src/behavior.ts [hunk 3]
```diff
@@ -97,6 +103,28 @@ export class SankeyDiagramBehavior implements IInteractiveBehavior {
...
+            this.clearSelection();
+
+            if (!sankeyDiagramUtils.areDataPointsSelected(this.selectedDataPoints, selectableDataPoints)) {
+                selectableDataPoints.forEach((subDataPoint: SelectableDataPoint) => {
+                    this.selectionHandler.handleSelection(subDataPoint, true);
+                });
+
+                this.selectedDataPoints = selectableDataPoints;
+            } else {
+                this.createAnEmptySelectedDataPoints();
```
Comment (MulyukovAidar) [inline src/behavior.ts:124]: why reset selected datapoints here?

**Comment by FirzinatKhuzeev** [comment]
Add description with screenshots

**Comment by s-ddavydenko** [comment]
> The screen reader output in not descriptive.
And outline is not prefect for this links example. Did you consider to use custom outline? (border or custom path element to highlight the current link)

@zBritva for now, this would be too difficult to implement

**Comment by s-ddavydenko** [comment]
> Could you add unit tests for the new feature?

Added

**Comment by s-ddavydenko** [comment]
> Please check if npm audit fix helps resolving audit issues (see npm audit). Also increase the visual version in pbiviz.json, package.json and add a release note in changelog.md

```
changed 4 packages, and audited 782 packages in 2s

109 packages are looking for funding
  run `npm fund` for details

# npm audit report

request  *
Severity: moderate
Server-Side Request Forgery in Request - https://github.com/advisories/GHSA-p8p7-x288-28g6
Depends on vulnerable versions of tough-cookie
No fix available
node_modules/request
  coveralls  *
  Depends on vulnerable versions of request
  node_modules/coveralls

tough-cookie  <4.1.3
Severity: moderate
tough-cookie Prototype Pollution vulnerability - https://github.com/advisories/GHSA-72xf-g2v4-qvf3
No fix available
node_modules/tough-cookie

3 moderate severity vulnerabilities

Some issues need review, and may require choosing
a different dependency.
```
Ran npm audit fix, this was the output

**Comment by s-ddavydenko** [comment]
> Add description with screenshots

added

**Comment by FirzinatKhuzeev** [comment]
> Looks good to me, but please, resolve conflicts.

done

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "powerbi-visuals-sankey",
-  "version": "3.1.3.0",
+  "version": "3.2.0",
```
Comment (uve) [inline package.json:3]: Return 4 digit version. It's mandatory for all CVs

File: pbiviz.json [hunk 5]
```diff
@@ -1,10 +1,10 @@
 {
   "visual": {
     "name": "SankeyDiagram",
-    "displayName": "Sankey 3.1.3.0",
+    "displayName": "Sankey 3.2.0",
```
Comment (uve) [inline pbiviz.json:4]: change to 3.2.0.0

**Review by uve — CHANGES_REQUESTED** [review]

File: pbiviz.json (hunk 5 above)
Comment (FirzinatKhuzeev) [reply pbiviz.json]: fixed

File: package.json (hunk 4 above)
Comment (FirzinatKhuzeev) [reply package.json]: done

**Pull request merged**